ql/src/java/org/apache/hadoop/hive/ql/parse/TaskCompiler.java

```diff
@@ -433,6 +434,12 @@ private String extractTableFullName(StatsTask tsk) throws SemanticException {
       return TaskFactory.get(columnStatsWork);
     } else {
       BasicStatsWork statsWork = new BasicStatsWork(tableScan.getConf().getTableMetadata().getTableSpec());
+      for (MapWork mapWork :  (Collection<MapWork>) currentTask.getMapWork()) {
+        if (mapWork.getAliasToPartnInfo() != null && mapWork.getAliasToPartnInfo().containsKey(table.getTableName())) {
```

Comment:
we have a full table object passed to the `StatsWork` constructor - what's wrong with that?